kubespawner/spawner.py

```diff
@@ -3048,14 +3068,39 @@ async def _load_profile(self, slug, selected_profile_user_options):
             # are in the form data posted. This prevents users who may be authorized
             # to only use one profile from being able to access options set for other
             # profiles
-            for user_selected_option_name in selected_profile_user_options.keys():
-                if (
-                    user_selected_option_name
-                    not in profile.get('profile_options').keys()
-                ):
-                    raise ValueError(
-                        f'Expected option {user_selected_option_name} for profile {profile["slug"]}, not found in posted form'
-                    )
+
+            for option_name, option in profile.get('profile_options').items():
+                if option_name not in selected_profile_user_options:
+                    # other_choice in Enabled:
+                    if option.get('other_choice', {}).get('enabled', False):
+                        if (
+                            f'{option_name}--other-choice'
+                            not in selected_profile_user_options
+                        ):
+                            raise ValueError(
+                                f'Expected option {option_name} for profile {profile["slug"]} or {option_name}--other-choice, not found in posted form'
+                            )
+                        other_choice = selected_profile_user_options[
+                            f'{option_name}--other-choice'
+                        ]
+
+                        # Validate value of 'other-choice' against validation regex
+                        if profile.get('profile_options')[option_name][
+                            'other_choice'
+                        ].get('validation_regex', False):
+                            other_choice_validation_regex = profile.get(
+                                'profile_options'
+                            )[option_name]['other_choice']['validation_regex']
+                            regex = re.compile(other_choice_validation_regex)
```

Comment:
You can directly call `re.match` here instead. If you look at the cpython source code, it gets compiled automatically (https://github.com/python/cpython/blob/904aef994262383ae916545908f0578c2d53cf31/Lib/re/__init__.py#L163). And compile also caches it (https://github.com/python/cpython/blob/904aef994262383ae916545908f0578c2d53cf31/Lib/re/__init__.py#L273) so it isn't recompiled unless necessary. Just having a .match here is cleaner than a compile and match.